crates/matrix-sdk/src/sliding_sync/mod.rs

```diff
@@ -1151,6 +1158,65 @@ mod tests {
         Ok(())
     }
 
+    #[async_test]
+    async fn test_room_subscriptions_are_reset_when_session_expires() -> Result<()> {
+        let (_server, sliding_sync) = new_sliding_sync(vec![SlidingSyncList::builder("foo")
+            .sync_mode(SlidingSyncMode::new_selective().add_range(0..=10))])
+        .await?;
+
+        let room_id_0 = room_id!("!r0:bar.org");
+        let room_id_1 = room_id!("!r1:bar.org");
+        let room_id_2 = room_id!("!r2:bar.org");
+
+        // Subscribe to two rooms.
+        sliding_sync.subscribe_to_rooms(&[room_id_0, room_id_1], None);
+
+        {
+            let sticky = sliding_sync.inner.sticky.read().unwrap();
+            let room_subscriptions = &sticky.data().room_subscriptions;
+
+            assert!(room_subscriptions.contains_key(&room_id_0.to_owned()));
```

Comment:
Is any of the `.to_owned()` necessary in that test? I thought the hashmap's key could be a `&RoomId` since it can be borrowed from an `OwnedRoomId` (the real key type).